common/safety-push-metrics/src/pusher.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::counters::Metrics;
+use libra_logger::*;
+use std::{env, sync::Arc, thread, thread::JoinHandle, time::Duration};
+
+const DEFAULT_PUSH_FREQUENCY_SECS: u64 = 15;
+
+pub struct MetricsPusher {
+    metrics: Arc<dyn Metrics + Send + Sync>,
+}
+
+impl MetricsPusher {
+    pub fn new(metrics: Arc<dyn Metrics + Send + Sync>) -> Self {
+        Self { metrics }
+    }
+
+    pub fn start(&'static self) -> JoinHandle<()> {
+        thread::spawn(move || {
+            // eg value for PUSH_METRICS_ENDPOINT: "http://pushgatewar.server.com:9091/metrics/job/safety_rules"
+            let push_metrics_endpoint = match env::var("PUSH_METRICS_ENDPOINT") {
```

Comment:
shouldn't this be in the metrics config -- especially since this happens on other systems? We could create a new subsection in metrics that defines struct PushMetricsConfig { ... }

Reply:
I would prefer to have fewer config files. Since we are only setting a couple of params (and I don't see this increasing in the future), having config files and a separate struct is an overkill.

Setting env variables is also fairly trivial when creating kubernetes specs and takes less effort than creating a config file.

Reply:
Commented inside the PR, but I think we should have this in config for simplicity and obviousness. Whether we want to support env variables...

cc: @mgorven  @sherry-x for their opinions